### PR TITLE
feature(playwright): Add selective launch for playwright

### LINKED ISSUE #498: Selective launch of autotests for the Playwright framework
At the moment, the possibility of selective launch is not implemented at the adapter level of the Playwright framework, so Allure TestsOps runs all tests.

### PR DESCRIPTION
**Pull request by vovsemenv**
### TestPlan usage

```diff
+ import { testPlanFilter } from "allure-playwright/dist/testplan";
export default {
  reporter: [
    [
      "allure-playwright",
    ],
  ],
  projects: [
    {
      name: "chromium",
    },
  ],
+  grep: testPlanFilter()
};
```

`testPlanFilter` function will read the test plan file and create `regexps` for each selected file

fixes #498
https://github.com/microsoft/playwright/issues/17037

### CODE REVIEW

**Comment by vovsemenv** [comment]
@pavelfeldman  or @mxschmitt please also check this PR and leave you opinion about using grep field for this task.

File: packages/allure-playwright/package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "typescript": "^4.4.2"
   },
   "dependencies": {
+    "@types/node": "^16",
```
Comment (mxschmitt) [inline packages/allure-playwright/package.json:35]: @playwright/test already depends on @types/node, so I'd move it to a devDependency instead.

File: packages/allure-playwright/src/testplan.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  version: string;
+  tests: {
+    id: number;
+    selector: string;
+  }[];
+}
+
+export const testPlanFilter = () => {
+  try {
+    const testPlanPath = process.env.ALLURE_TESTPLAN_PATH;
```
Comment (baev) [inline packages/allure-playwright/src/testplan.ts:13]: move test plan parsing to commons

**Review by baev — CHANGES_REQUESTED** [review]

File: packages/allure-playwright/src/testplan.ts (hunk 2 above)
Comment (baev) [reply packages/allure-playwright/src/testplan.ts:13]: @vovsemenv see https://github.com/allure-framework/allure-js/pull/464

File: packages/allure-playwright/test/testplan.spec.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    (writer) => {
+      return writer.tests;
+    },
+    {},
+    {
+      ALLURE_TESTPLAN_PATH: testPlanFilename,
+    },
+  );
+
+  expect(results.length).toBe(3);
```
Comment (epszaw) [inline packages/allure-playwright/test/testplan.spec.ts:63]: Could you check what the tests exactly we have there?
Current check is a bit vague and we potentially can have any results, even not matched by filter function.

File: packages/allure-playwright/src/testplan.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    return selectedTests;
+  } catch (e) {
+    return undefined;
+  }
+};
+
+const reRegExpChar = /[\\^$.*+?()[\]{}|]/g,
+  reHasRegExpChar = RegExp(reRegExpChar.source);
+
+export const escapeRegExp = (value: string): string => {
```
Comment (epszaw) [inline packages/allure-playwright/src/testplan.ts:40]: It would be nice, to test utility functions too

**Review by epszaw — CHANGES_REQUESTED** [review]
✅ The functionality works well, tested locally
✅ The code looks also good
⚠️ Please, pay your attention to my comments about tests

**Review by epszaw — APPROVED** [review]
✅ Looks much better! Thank you for the changes!

**Review by baev — APPROVED** [review]

**Pull request merged**